cli/lsp/diagnostics.rs

```diff
@@ -296,57 +301,72 @@ struct ChannelUpdateMessage {
   batch_index: Option<usize>,
 }
 
-#[derive(Clone, Debug)]
+#[derive(Debug)]
 struct SpecifierState {
-  has_no_cache_diagnostic: bool,
+  version: Option<i32>,
+  no_cache_diagnostics: Vec<lsp::Diagnostic>,
 }
 
 #[derive(Clone, Debug, Default)]
 pub struct DiagnosticsState {
-  specifiers: HashMap<ModuleSpecifier, (Option<i32>, SpecifierState)>,
+  specifiers: Arc<RwLock<HashMap<ModuleSpecifier, SpecifierState>>>,
```

Comment:
I think we can just make this a parking_lot::rwlock since these method calls will be very short lived? Then we could get rid of the `async` on all the methods.